framework/src/problems/FEProblemBase.C

```diff
@@ -6359,6 +6359,23 @@ FEProblemBase::addOutput(const std::string & object_type,
   if (common)
     parameters.applyParameters(*common, exclude);
 
+  // If file_base parameter has not been set with the common parameters, we inquire app to set it
+  // here
+  if (parameters.have_parameter<std::string>("file_base") &&
```

Comment:
This logic really shouldn't be in FEProblemBase. It should be in the Outputs system somewhere. How about creating a little utility in the Output system to do this naming that you've put here.

Reply:
Oh I see, I see your new comment above in the FileBase object about MOOSE setting things for you. This is where that is occurring. I don't think this is the best encapsulation though. Once again, we should be putting all of this logic in one place. Either the Outputs system (maybe FileBase), or MooseApp. If we are a MultiApp we still could query that in either of those other two places to get good encapsulation in making these naming decisions in one place.